cmd/auth/main.go

```diff
@@ -1,115 +1,34 @@
-		Access token of user
-	@SecurityRequirement access_key
-
-	@Tag User
-		User information
-	@Tag Forgot Password
-		Password recovery process
-	@Tag Administration
-		Administrative operation
-	@Tag SSO
-		Single sign-on
-*/
 func main() {
-	// logging initialization
-	logging.SetModule("auth")
-	loggerFactory := logging.NewFactory(
-		logging.NewDefaultLogHook(nil),
-		&sentry.LogHook{Hub: sentry.DefaultClient.Hub},
-	)
-	logger := loggerFactory.NewLogger("auth")
-
-	envErr := godotenv.Load()
-	if envErr != nil {
-		logger.WithError(envErr).Debug("Cannot load .env file")
-	}
-
-	configuration := configuration{}
-	envconfig.Process("", &configuration)
-	if configuration.ValidHosts == "" {
-		configuration.ValidHosts = configuration.Host
-	}
-
-	validator := validation.NewValidator("http://v2.skgyear.io")
-	validator.AddSchemaFragments(
-		oauthhandler.ChallengeRequestSchema,
-	)
-
-	dbPool := db.NewPool()
-	redisPool, err := redis.NewPool(configuration.Redis)
+	err := godotenv.Load()
 	if err != nil {
-		logger.Fatalf("fail to create redis pool: %v", err.Error())
-	}
-	asyncTaskExecutor := async.NewExecutor(dbPool)
-	var assetGearLoader *template.AssetGearLoader
-	if configuration.Template.AssetGearEndpoint != "" && configuration.Template.AssetGearMasterKey != "" {
-		assetGearLoader = &template.AssetGearLoader{
-			AssetGearEndpoint:  configuration.Template.AssetGearEndpoint,
-			AssetGearMasterKey: configuration.Template.AssetGearMasterKey,
-		}
+		log.Fatalf("failed to load .env file: %s", err)
 	}
 
-	var reservedNameChecker *loginid.ReservedNameChecker
-	reservedNameChecker, err = loginid.NewReservedNameChecker(configuration.ReservedNameSourceFile)
+	serverCfg, err := config.LoadServerConfigFromEnv()
 	if err != nil {
-		logger.Fatalf("fail to load reserved name source file: %v", err.Error())
+		log.Fatalf("failed to load server config: %s", err)
 	}
 
-	authDependency := auth.DependencyMap{
-		EnableFileSystemTemplate: configuration.Template.EnableFileLoader,
-		AssetGearLoader:          assetGearLoader,
-		AsyncTaskExecutor:        asyncTaskExecutor,
-		UseInsecureCookie:        configuration.UseInsecureCookie,
-		StaticAssetURLPrefix:     configuration.StaticAssetURLPrefix,
-		DefaultConfiguration:     configuration.Default,
-		Validator:                validator,
-		ReservedNameChecker:      reservedNameChecker,
+	rootDeps, err := deps.NewRootContainer(serverCfg)
+	if err != nil {
+		log.Fatalf("failed to setup server: %s", err)
 	}
 
-	task.AttachPwHousekeeperTask(asyncTaskExecutor, authDependency)
-	task.AttachSendMessagesTask(asyncTaskExecutor, authDependency)
-
-	router := setupRoutes(configuration, dbPool, redisPool, authDependency)
-
-	srv := &http.Server{
-		Addr:    configuration.Host,
-		Handler: router,
-	}
-	server.ListenAndServe(srv, logger, "Starting auth gear")
+	server := httputil.NewServer(rootDeps.LoggerFactory, &http.Server{
+		Addr:    serverCfg.ListenAddr,
+		Handler: nil,
```

Comment:
Set the handler?

Reply:
Will update it after fixing the dependencies.